root.c

```diff
@@ -2083,6 +2083,32 @@ bool w_root_stop_watch(w_root_t *root)
   return stopped;
 }
 
+json_t *w_root_stop_watch_all(void)
+{
+  w_ht_iter_t iter;
+  json_t *stopped = json_array();
+
+  pthread_mutex_lock(&root_lock);
+  if (w_ht_first(watched_roots, &iter)) do {
+    w_root_t *root = w_ht_val_ptr(iter.value);
+    w_string_t *path = root->root_path;
+    json_t *json_root = json_string_binary(path->buf, path->len);
+
+    if (w_ht_iter_del(watched_roots, &iter)) {
```

Comment:
there's a potential race here; root may have its final ref released by the delete operation, making it unsafe to call w_root_cancel on it below.  You'll need to `w_root_addref` prior to calling this, and then `w_root_delref` before the next iteration